### PR TITLE
Fix for issue#897.

### PR DESCRIPTION
**Pull request by bbergquist0930**
Added starting of the request timer even when the prvPublishMmessage returns a failure.

This is a fix for [Issue#897](https://github.com/aws/amazon-freertos/issues/897)

Description
-----------
Added the starting of the request timer even if the "prvPublishMessage" returns a failure.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by sarenameas** [comment]
Hello,
Thanks for bringing this up. This intermittent issue we have seen and are actively working on internally. We will get back to you in a week or so after we recreate it internally and try your fix with a few days of soak as well.

File: libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c [hunk 1]
```diff
@@ -1258,7 +1258,9 @@ static OTA_Err_t prvPublishGetStreamMessage( OTA_FileContext_t * C )
                     if( eResult != IOT_MQTT_SUCCESS )
                     {
                         OTA_LOG_L1( "[%s] Failed: %s\r\n", OTA_METHOD_NAME, pcTopicBuffer );
-                        /* Don't return an error. Let max momentum catch it since this may be intermittent. */
+                        /* Don't return an error. Let max momentum catch it since this may be intermittent.
```
Comment (pvyawaha) [inline libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c]: Can we please move the call to prvStartRequestTimer after if-else block as it is being called in both cases and the if-else blocks will be left for logs only.

Comment (bbergquist0930) [reply libraries/freertos_plus/aws/ota/src/aws_iot_ota_agent.c]: Updated the change to only restart the timer after the if-else block and added a comment on the logic of why it is always restarted.

**Pull request merged**